### ISSUE TITLE
ConvTranspose2d + autograd.grad + backward causes "RuntimeError: cuDNN requires contiguous weight tensor"

### ISSUE THREAD

**Issue by WarBean** [pytorch/pytorch#5044]
Minimal test script:

```
import torch
import torch.nn as nn
from torch.autograd import Variable

a = Variable(torch.rand(10, 3, 5, 5).cuda(), requires_grad = True)
deconv = nn.ConvTranspose2d(3, 10, 3, 1, 1).cuda()
b = deconv(a)
c = b.mean()
g = torch.autograd.grad(outputs = c, inputs = a, retain_graph = True, create_graph = True, only_inputs = True)[0]
loss = g.mean()
loss.backward()
```

Error message:

```
Traceback (most recent call last):
  File "backward_grad_error.py", line 11, in <module>
    loss.backward()
  File "/mnt/lustre/zhenghuabin/anaconda3/envs/py35/lib/python3.5/site-packages/torch/autograd/variable.py", line 167, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph, retain_variables)
  File "/mnt/lustre/zhenghuabin/anaconda3/envs/py35/lib/python3.5/site-packages/torch/autograd/__init__.py", line 99, in backward
    variables, grad_variables, retain_graph)
RuntimeError: cuDNN requires contiguous weight tensor
```

**Comment by WarBean** [comment]
Temporal solution：replacing deconv with upsample + conv is ok:

```
import torch
import torch.nn as nn
from torch.autograd import Variable

a = Variable(torch.rand(10, 3, 5, 5).cuda(), requires_grad = True)
deconv = nn.ConvTranspose2d(3, 10, 3, 1, 1).cuda()
upsample = nn.Upsample(scale_factor = 2, mode = 'nearest').cuda()
conv = nn.Conv2d(3, 10, 3, 1, 1).cuda()

#b = deconv(a)
b = conv(upsample(a))
c = b.mean()
g = torch.autograd.grad(outputs = c, inputs = a, retain_graph = True, create_graph = True, only_inputs = True)[0]
loss = g.mean()
loss.backward()
```

Though the original bug is not fixed.

**Comment by apaszke** [comment]
cc @ezyang

**Comment by SsnL** [comment]
I can't repro it on master. It might have been fixed already. 

Could you submit issues according to the autopopulated template please? It will make debugging much easier, especially in this case since cudnn cuda and torch versions all matter. @WarBean

**Comment by ezyang** [comment]
This was fixed in bc7a41af7d541e64f8b8f7318a7a2248c0119632

**Comment by WarBean** [comment]
@ezyang Thanks.
